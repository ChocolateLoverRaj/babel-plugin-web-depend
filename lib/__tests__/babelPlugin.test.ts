import { transformSync } from '@babel/core'
import babelPluginWebDepend from '../babelPlugin'
import never from 'never'

const transform = (code: string): string =>
  transformSync(code, { plugins: [[babelPluginWebDepend]] })?.code ?? never('No code')

test('import as namespace', () => {
  expect(transform("import * as React from 'react'")).toMatchSnapshot()
})

test('import default', () => {
  expect(transform("import React from 'react'")).toMatchSnapshot()
})

describe('import named', () => {
  describe('single import', () => {
    test('no rename', () => {
      expect(transform("import { useState } from 'react'")).toMatchSnapshot()
    })

    test('rename', () => {
      expect(transform("import { useState as useStateReact } from 'react'")).toMatchSnapshot()
    })
  })

  describe('multiple imports', () => {
    test('no renames', () => {
      expect(transform("import { useState, useEffect } from 'react'")).toMatchSnapshot()
    })

    test('renames', () => {
      expect(transform(
        "import { useState as useStateReact, useEffect as useEffectReact } from 'react'"))
        .toMatchSnapshot()
    })

    test('mixed', () => {
      expect(transform("import { useState, useEffect as useEffectReact } from 'react'"))
        .toMatchSnapshot()
    })
  })
})

test('import default and named', () => {
  expect(transform("import React, { useState } from 'react'")).toMatchSnapshot()
})

test('multiple imports', () => {
  expect(transform(`
    import * as React from 'react'
    import cn from 'classnames'
  `)).toMatchSnapshot()
})

test('dynamic import', () => {
  expect(transform("import('boolean-to-signed')")).toMatchSnapshot()
})
