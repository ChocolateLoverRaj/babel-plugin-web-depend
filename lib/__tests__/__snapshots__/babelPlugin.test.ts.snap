// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynamic import 1`] = `
"import(\\"https://todo-web-depend-path\\").then(async ({
  default: importDepend
}) => await importDepend('boolean-to-signed'));"
`;

exports[`import as namespace 1`] = `
"import _importDepend from \\"https://todo-web-depend-path\\";
const React = await _importDepend('react');"
`;

exports[`import default 1`] = `
"import _importDepend from \\"https://todo-web-depend-path\\";
const {
  default: React
} = await _importDepend('react');"
`;

exports[`import default and named 1`] = `
"import _importDepend from \\"https://todo-web-depend-path\\";
const {
  default: React,
  useState: useState
} = await _importDepend('react');"
`;

exports[`import named multiple imports mixed 1`] = `
"import _importDepend from \\"https://todo-web-depend-path\\";
const {
  useState: useState,
  useEffect: useEffectReact
} = await _importDepend('react');"
`;

exports[`import named multiple imports no renames 1`] = `
"import _importDepend from \\"https://todo-web-depend-path\\";
const {
  useState: useState,
  useEffect: useEffect
} = await _importDepend('react');"
`;

exports[`import named multiple imports renames 1`] = `
"import _importDepend from \\"https://todo-web-depend-path\\";
const {
  useState: useStateReact,
  useEffect: useEffectReact
} = await _importDepend('react');"
`;

exports[`import named single import no rename 1`] = `
"import _importDepend from \\"https://todo-web-depend-path\\";
const {
  useState: useState
} = await _importDepend('react');"
`;

exports[`import named single import rename 1`] = `
"import _importDepend from \\"https://todo-web-depend-path\\";
const {
  useState: useStateReact
} = await _importDepend('react');"
`;

exports[`multiple imports 1`] = `
"import _importDepend from \\"https://todo-web-depend-path\\";
const React = await _importDepend('react');
const {
  default: cn
} = await _importDepend('classnames');"
`;
